src/state_machine.zig

```diff
@@ -437,19 +444,19 @@ pub fn StateMachineType(comptime Storage: type) type {
                                 count += 1;
                             }
                         } else {
-                            assert(result == .linked_event_failed);
+                            assert(result == .linked_event_failed or result == .linked_event_chain_open);
```

Comment:
I added a new test "linked_event_chain_open for an already failed batch" that would trip this assert without the `or`.